pkg/operator/ceph/cluster/mgr/mgr.go

```diff
@@ -591,3 +612,43 @@ func applyMonitoringLabels(c *Cluster, serviceMonitor *monitoringv1.ServiceMonit
 		}
 	}
 }
+
+func shouldRotateMgrKeys(c *clusterd.Context, clusterInfo *cephclient.ClusterInfo) (bool, error) {
+	clusterObj := &cephv1.CephCluster{}
+	if err := c.Client.Get(clusterInfo.Context, clusterInfo.NamespacedName(), clusterObj); err != nil {
+		return false, errors.Wrapf(err, "failed to get cluster %v.", clusterInfo.NamespacedName())
+	}
+	desiredCephVersion := clusterInfo.CephVersion
+	// TODO: for rotation WithCephVersionUpdate fix this to have the right runningCephVersion and desiredCephVersion
+	runningCephVersion := clusterInfo.CephVersion
+
+	shouldRotateKeys, err := keyring.ShouldRotateCephxKeys(clusterObj.Spec.Security.CephX.Daemon, runningCephVersion, desiredCephVersion, *clusterObj.Status.Cephx.Mgr)
+	if err != nil {
+		return false, errors.Wrap(err, "failed to check if mgr daemon keys should be rotated or not")
+	}
+
+	return shouldRotateKeys, nil
+}
+
+func updateMgrCephxStatus(c *clusterd.Context, clusterInfo *cephclient.ClusterInfo, didRotate bool) error {
+	err := retry.RetryOnConflict(retry.DefaultRetry, func() error {
+		cluster := &cephv1.CephCluster{}
+		if err := c.Client.Get(clusterInfo.Context, clusterInfo.NamespacedName(), cluster); err != nil {
```

Comment:
```suggestion
		if err := c.Client.Get(clusterInfo.Context, clusterInfo.NamespacedName(), cluster); err != nil {
```
can we move `get` call out of retry loop? at least I'm moving in case of CSI rotation not sure if that applicable here also.

Reply:
But why?  I think if there is conflict, then it means we don't have the updated copy of the CepCluster. So we should be querying for the cephCluster again inside the retry loop to get the latest copy of the resource.